src/lexer.coffee

```diff
@@ -242,6 +242,22 @@ exports.Lexer = class Lexer
     number = match[0]
     lexedLength = number.length
 
+    # Property number notation, e.g. a.3
+    prev = @prev()
+    if prev and not prev.spaced
+      if prev[0] in [CALLABLE..., '?.', '::', '?::']
+        s = 0
+        nProp = number
+        if nProp[0] is "."
+          @token '.', '.', s++, 1
+          nProp = nProp[1..]
+
+        for num, x in nProp.split "."
+          @token '.', '.', s++, 1 if (x+1) % 2 is 0
```

Comment:
@zdenko since `NUMBER` will match at most one `.`, the `(x+1) % 2 is 0` logic is overly generic and therefore hard to follow. Should work just to say `if x is 1`